modules/saas/apps/saml-saas/src/main/java/com/liferay/saml/saas/internal/portlet/filter/SamlAdminRenderFilter.java

```diff
@@ -62,12 +64,13 @@ public void doFilter(
 
 		chain.doFilter(renderRequest, renderResponse);
 
-		String generalTab = ParamUtil.getString(
-			renderRequest, "tabs1", "general");
+		String tabs1 = ParamUtil.getString(renderRequest, "tabs1", "general");
 		String mvcrenderCommandName = ParamUtil.getString(
 			renderRequest, "mvcRenderCommandName", null);
 
-		if (!generalTab.equals("general") || (mvcrenderCommandName != null)) {
+		if (!Objects.equals(tabs1, "general") ||
+			(mvcrenderCommandName != null)) {
```

Comment:
Just wondering if this is incorrect? The DefaultViewMVCRenderCommand name doesn't need to be specified with the "mvcRenderCommandName" parameter because it is the default MVCRenderCommand for the portlet. Surprised it works now actually!

Also I think the camel casing is wrong for the variable?

cc @csierra @martamedio

Reply:
Sorry, realize now that the condition is for the filter not to do anything when it's != null. So we might have the reverse problem, that if the URL specifies "/admin" then it will not do anything, but logically it's the same DefaultViewMVCRenderCommand

Reply:
There are several cases to check with this so the button appears or not, and the easiest way to decide is to use a combination of these two parameters:

- When you enter from the main menu (_SAML Admin_): button must be shown (`tab1 = null/general && mvcRenderCommandName = null`)
- When you open a certificate popup: button must not be shown inside (`tab1 = "general" && mvcRenderCommandName = "/admin/updateCertificate"`)
- When you're in another tab: button must not be shown (`tab1 != null/general`)
- When you try to add a New Identity Provider: button must not be shown (`tab = null && mvcRenderCommandName = /admin/edit_identity_provider_connection`)
- When you submit the "_Export_" button: button must be shown again (`tab1 = null/general && mvcCommandRenderName = "/admin"`)